### PR TITLE
feat: Save sql chart in space and fix slug for sql charts

### PR DESCRIPTION
**Pull request by rephus**
Closes:  #10779

#### existing space

Screenshot from 2024-07-19 16-35-30
Screenshot from 2024-07-19 16-34-55
#### new space

Screenshot from 2024-07-19 16-47-58
Screenshot from 2024-07-19 16-48-06
Screenshot from 2024-07-19 16-48-13

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/ResourceView/resourceUtils.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const getResourceUrl = (projectUuid: string, item: ResourceViewItem) => {
             return `/projects/${projectUuid}/dashboards/${item.data.uuid}/view`;
         case ResourceViewItemType.CHART:
             if (item.data.source === ChartSourceType.SQL) {
-                return `/projects/${projectUuid}/sql-runner-new/saved/${item.data.uuid}`;
+                return `/projects/${projectUuid}/sql-runner-new/saved/${item.data.slug}`;
```
Comment (joaoviana) [inline packages/frontend/src/components/common/ResourceView/resourceUtils.ts:59]: 💪

File: packages/common/src/types/savedCharts.ts [hunk 2]
```diff
@@ -594,6 +594,7 @@ export type ChartSummary = Pick<
     | 'pinnedListUuid'
     | 'dashboardUuid'
     | 'dashboardName'
+    | 'slug'
```
Comment (joaoviana) [inline packages/common/src/types/savedCharts.ts:597]: do we need to run a `generate-api` here?

File: packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx [hunk 3]
```diff
@@ -76,17 +75,25 @@ export const SaveSqlChartModal: FC<Props> = ({ opened, onClose }) => {
         if (spaces.length === 0) {
             return;
         }
+        let newSpace = form.values.newSpaceName
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx:78]: this is great, but we should set the `initial space` on the `useEffect` where we set the initial name & description.

Something that we have on `SaveToSpaceOrDashboard`

```tsx

 let initialSpaceUuid;

            const isValidDefaultSpaceUuid = spaces.some(
                (space) => space.uuid === defaultSpaceUuid,
            );

            if (spaces && spaces.length > 0) {
                initialSpaceUuid = isValidDefaultSpaceUuid
                    ? defaultSpaceUuid
                    : spaces[0].uuid;
            }
```

could you add something like this?

**Review by joaoviana — CHANGES_REQUESTED** [review]
Looks great! Just one bit on adding the initial space uuid so the user has a space already selected 👍

File: packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx (hunk 3 above)
Comment (rephus) [reply packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx:81]: updated

**Comment by rephus** [comment]
I had to change a few models to include `slug`

File: packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx [hunk 4]
```diff
@@ -51,7 +50,10 @@ export const SaveSqlChartModal: FC<Props> = ({ opened, onClose }) => {
         if (!form.values.description && description) {
             form.setFieldValue('description', description);
         }
-    }, [name, form, description]);
+        if (!form.values.spaceUuid && spaces.length > 0) {
+            form.setFieldValue('spaceUuid', spaces[0].uuid);
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx:54]: great!

**Review by joaoviana — APPROVED** [review]
Just going to test in Render but approving in the meantime 👍

**Comment by joaoviana** [comment]
All good @rephus, `slug` resource item link works 💪

**Pull request merged**